### DISCUSSION TITLE
Black Screen which says macOS is upgrading disappears

### DISCUSSION THREAD

**Discussion by rbahena2** [grahampugh/erase-install#283] [General]
![PXL_20220827_213654530](https://user-images.githubusercontent.com/112286316/187052816-e6265f41-8294-4c6b-9b86-8a35de5117bc.jpg)
Hi, I noticed recently when using erase-install the user is asked to put in their password on Silicon Mac and once the password is accepted the full screen black screen says "Please wait as we prepare your computer for upgrading macOS...". Everything is working perfectly so far.
After about 10-14 minutes the black screen just closes and desktop is presented again and I can use the computer for about 1-3 minutes before an abrupt reboot happens and the upgrade is underway. I have no rebootdelays in the command line. This is happening on all macs I try this on. Anyone else experience this? Is there a work around?
i notice in erase-install.log it says "attempting to terminate the 'jamfhelper' process - Termination message indicates success" right after the black screen closes. As you can see from screenshot I can still use computer after it completes.

**Comment by hcstechgroup** [comment]
Graham, I am having the same issue trying to upgrade to macOS Monterey 12.5.1 on an M1 MacBook Air.  This did not happen in the past.  Not sure if 12.5.1 has anything to do with it.  Any thoughts?

**Comment by hcstechgroup** [comment]
More on this.  The black macOS is upgrading screen goes way after 10 minutes or so and asks me for my password a second time.  Once I enter it in, the black screen comes back for a few seconds, then goes away again.  Now it’s sitting at the finder for almost 10 minutes.  I will let it sit for a while and report back.

**Comment by hcstechgroup** [comment]
I let it sit at the finder for over 30 minutes.  nothing happened.  below is the tail end of the  log file.  I manually rebooted and the upgrade did not complete.

Preparing: 98.3%            
Preparing: 98.8%            
Preparing: 99.9%            
Preparing: 100.0%            
   [erase-install] attempting to terminate the 'Self Service' process - Termination message indicates success


   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
   [erase-install] ERROR: 'caffeinate' could not be killed


   [erase-install] attempting to terminate the 'jamfHelper' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1078: 31475 Terminated: 15          "$jamfHelper" -windowType $window_type -title "${!dialog_reinstall_title}" -heading "${!dialog_reinstall_heading}" -description "${!dialog_reinstall_desc}" -icon "$dialog_reinstall_icon"


   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
   [erase-install] ERROR: 'caffeinate' could not be killed



Helper tool crashed...

Preparing: 100%          
Signaling PID: 31356 that we are ready to restart.

Restarting...

**Reply by grahampugh** [reply]
`ERROR: 'caffeinate' could not be killed` - that's a new one. Can you reproduce that on more than one computer?

**Reply by hcstechgroup** [reply]
I only have one test Mac at the moment but I'm working in tandem with Rene, the guy who started this discussion.  He was able to comment out the jamf helper part of the script on line 1571 and that worked for both of us.  Hope that helps.

**Reply by grahampugh** [reply]
What version of erase-install? Line numbers change :)

**Reply by hcstechgroup** [reply]
version 26.2

**Reply by rbahena2** [reply]
<img width="554" alt="Screen Shot 2022-08-29 at 11 44 46" src="https://user-images.githubusercontent.com/112286316/187284102-af2e9303-24a8-4eff-9776-a2cf086d8131.png">

**Reply by rbahena2** [reply]
Line 1571 is commented out and I repackaged the software and it works great now. Not sure if there are any gotchas by commenting out that line though.

**Reply by grahampugh** [reply]
Presumably if you were using jamfHelper, it wouldn't be good...

27.0 has some changes to that section anyway. I'm slowly starting some tests on this but it'll take a while.

**Comment by grahampugh** [comment]
@rbahena2 we've always struggled with that gap at the end. I thought it had been solved a few versions back, but will have to do some more tests.

**Reply by rbahena2** [reply]
Oh ok, good to know. BTW, awesome tool you have here. We love it.
